### ISSUE TITLE
Sphere loses detail after coordinates exceed 2^16

### ISSUE THREAD

**Issue by gc625** [pyvista/pyvista#6685]
### Describe the bug, what's wrong, and what you expected.

I'm planning to use pyvista for visualizing 3d objects which may have very large coordinates due to the coordinate system I am using. I found that when shapes are too far from the origin (approx beyond (2^16,2^16,2^16)), the geometries start losing detail. I suspect its because of floating point precision? What is the prefered way of handling large coordinates?

### Steps to reproduce the bug.

```python
import pyvista as pv
import numpy as np
n = 16
# Create the sphere with double-precision points
sphere = pv.Sphere(radius=1.0, center=(2**n, 2**n, 2**n))
# this does nothing
sphere.points = sphere.points.astype(np.float64)

# Create the plotter and add the sphere
plotter = pv.Plotter()
plotter.add_mesh(sphere, color='red', opacity=0.6, show_edges=True, label='Sphere')

# Show the plot
plotter.show()

```

### System Information

```shell
--------------------------------------------------------------------------------
  Date: Wed Oct 02 12:56:20 2024 HKT

                OS : Linux
            CPU(s) : 24
           Machine : x86_64
      Architecture : 64bit
               RAM : 31.1 GiB
       Environment : Python
       File system : ext4
        GPU Vendor : NVIDIA Corporation
      GPU Renderer : NVIDIA GeForce RTX 3080/PCIe/SSE2
       GPU Version : 4.5.0 NVIDIA 535.183.01
  MathText Support : False

  Python 3.8.10 (default, Sep 11 2024, 16:02:53)  [GCC 9.4.0]

           pyvista : 0.44.1
               vtk : 9.3.1
             numpy : 1.24.4
        matplotlib : 3.1.2
            scooby : 0.10.0
             pooch : 1.8.2
            pillow : 10.0.1
           imageio : 2.31.3
             PyQt5 : 5.14.1
           IPython : 8.12.2
        ipywidgets : 8.1.0
             scipy : 1.10.1
              tqdm : 4.66.1
            meshio : 5.3.5
      nest_asyncio : 1.5.6
--------------------------------------------------------------------------------
```

### Screenshots

![image](https://github.com/user-attachments/assets/2f441f8b-c9a3-49cd-8ec6-839ba8b61a0d)
![image](https://github.com/user-attachments/assets/00b36847-b019-4d3d-b542-bdde8387ddbb)
![image](https://github.com/user-attachments/assets/246fea73-83ff-4887-87e0-4661f2a4ccde)
![image](https://github.com/user-attachments/assets/cc4ebb35-d5c0-4a6c-a6ae-012b24cf0929)
and at `n=24` it just degenerates to a cube
![image](https://github.com/user-attachments/assets/99467e80-9aeb-408d-95f9-5a54f891ae5c)

**Comment by MatthewFlamm** [comment]
In this case, I would calculate my position in a new coordinate system so that the position does not create problems with precision.  Given that the object has a radius of 1, why do you need to place the object at (2^24, 2^24, 2^24)?  Why not place it at (0, 0, 0) and mentally account that this is actually at (2^24, 2^24, 2^24)?

**Comment by banesullivan** [comment]
This seems to have come up a lot over time. Most notably on my mind was https://github.com/pyvista/pyvista/issues/5690 and other threads where dealing with geospatial coordinates where the level of detail is off in the the 8/9th decimal place which is lost on float32

**Comment by banesullivan** [comment]
>I found that when shapes are too far from the origin (approx beyond (2^16,2^16,2^16)), the geometries start losing detail. I suspect its because of floating point precision? What is the prefered way of handling large coordinates?

Yeah the best thing to do is to translate to some reference coordinate system (subtract the mean)
